### PR TITLE
Show correct status when Licensed but Cancelled

### PR DESCRIPTION
**Pull request by alex-deaconu**
If a Product was cancelled or suspended, the status shows as red
Update the gadget status when it is in the licensed state

Refactor code by splitting into directives and filters

### CODE REVIEW

File: web/scripts/editor/filters/ftr-subscription-status-message.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            oneDay)));
+        };
+
+        var _getMessage = function () {
+          var statusMessage = gadget.subscriptionStatus;
+          if (gadget.isLicensed) {
+            statusMessage = $filter('translate')
+              ('editor-app.subscription.status.professional');
+          } else if (gadget.subscriptionStatus === 'Not Subscribed' && gadget.trialPeriod > 0) {
+            statusMessage = statusMessage + ' - ' + gadget.trialPeriod +
```
Comment (olegrise) [inline web/scripts/editor/filters/ftr-subscription-status-message.js]: @alex-deaconu do you think this message is outdated considering that we don't sell widgets separately anymore? Should it be "For Licensed Displays" instead?

Comment (alex-deaconu) [reply web/scripts/editor/filters/ftr-subscription-status-message.js]: @olegrise I tend to agree. To confirm, we would show the "For Licensed Displays" message for all cases except for when they're Subscribed directly to the Widget. While the message is the same, the colour would be dependent on whether they have a Plan (Green) or not (Red).
@ezequielc FYI. Thoughts? Thanks.

Comment (ezequielc) [reply web/scripts/editor/filters/ftr-subscription-status-message.js]: @alex-deaconu @olegrise Good suggestion. Thanks

**Comment by alex-deaconu** [comment]
@Rise-Vision/apps implemented as Oleg suggested inline, please review last 2 commits. Thanks!

**Comment by olegrise** [comment]
@alex-deaconu There is an inconsistency in case when I don't have a plan - the status message "For Licensed Displays" is red in placeholder playlist and green on the modal. BTW, it seems it was like that before your change, so I don't know if it's expected behaviour or not.

inconsistency

**Comment by alex-deaconu** [comment]
@olegrise I did not make any changes to the classes of either of those elements. In fact the only states that are `red` in the Playlist Item Modal are defined in the CH css (cancelled, expired, suspended):
https://github.com/Rise-Vision/common-header/blob/master/src/scss/ui-components/subscription.scss#L30

Whereas the class `.editor-playlist-item .playlist-item-status` is always present in the Placeholder Playlist Item, so it will always be red whenever that notice is shown (when not subscribed and not licensed).

I don't think we need to change this with this fix. @ezequielc thoughts? Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu I agree with you, I don't think we need to change it with this fix, even though it is a weird styling. Thanks

**Comment by olegrise** [comment]
@ezequielc @alex-deaconu sounds good. Thanks for confirming!

**Pull request merged**